charts/ceph-csi-rbd/templates/nodeplugin-clusterrole.yaml

```
@@ -15,7 +15,8 @@ rules:
     resources: ["nodes"]
     verbs: ["get"]
 {{- end }}
+  # allow to read Vault Token and connection options from the Tenants namespace
   - apiGroups: [""]
-    resources: ["secrets"]
+    resources: ["secrets", "configmaps"]
```

This should go as a new resource. adding to the existing resource might  give more access when we add access to only one resouce

```suggestion
- apiGroups: [""]
  resources: ["secrets"]
  verbs: ["get"]
- apiGroups: [""]
  resources: ["configmaps"]
  verbs: ["get"]
```

---

How do you mean _adding to the existing resource might give more access_ ? The comment above clearly states that these two resources are needed for the Vault Tokens functionality. If these resources/verbs are changed in a future PR, we'll see the comment and can judge if the change is valid or needs to be split out.

---

Yes if we are updating verb for only configmap later to watch or something like that we need to split the resources and add a new one for configmap, we can add separate resources now only so that we don't need to worry about splitting later

---

currently, we are keeping each resource independently in both helm and kubernetes yaml files

---

> currently, we are keeping each resource independently in both helm and kubernetes yaml files

This is the most compelling reason to do so now. Rules should be documented with the reason why it is needed, but as that is not done yet everywhere, splitting seems okay-ish.